apps/system/js/app_window.js

```diff
@@ -1161,23 +1156,37 @@
         case 'application-name':
           // Apps have a compulsory name field in their manifest
           // which takes precedence.
-          if (!this.isBrowser()) {
+          if (this.manifestURL || this.webManifestURL) {
             return;
           }
-          this.updateName(detail.content);
+          this.name = detail.content;
           this.publish('namechanged');
           break;
       }
-
     };
 
   AppWindow.prototype._handle_mozbrowsermanifestchange =
     function aw__handle_mozbrowsermanifestchange(evt) {
-      if (evt.detail.href) {
-        this.webManifestURL = evt.detail.href;
+      if (!evt.detail.href) {
+        return;
       }
+      this.webManifestURL = evt.detail.href;
+      this._fetchManifest(this.webManifestURL);
     };
 
+  AppWindow.prototype._fetchManifest = function aw_fetchManifest(manifestURL) {
+    return WebManifestHelper.getManifest(manifestURL)
```

Comment:
nit: could we lazy load webManifestHelper here? We don't need it on startup I think.

return LazyLoader.load('shared/js/web_manifest_helper.js').then(() => {
  return WebManifestHelper.getManifest(manifestURL);
}).then((webManifest) => {
  // ....
});